### PR TITLE
qemu.tests: Add ovs_host_vlan test case and script.

### PR DESCRIPTION
**Pull request by FengYang**
ID: 1081112

Signed-off-by: Feng Yang <fyang@redhat.com>

### CODE REVIEW

File: qemu/tests/ovs_host_vlan.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                                               username, username,
+                                               file_path, file_path,
+                                               d_port=data_port,
+                                               timeout=2,
+                                               check_sum=True,
+                                               s_session=sessions[0],
+                                               d_session=sessions[1],
+                                               file_transfer_timeout=f_timeout)
+    finally:
+        ovs_del_cmd = "ovs-vsctl del-br %s" % bridge_name
```
Comment (xutian) [inline qemu/tests/ovs_host_vlan.py]: Do we need to stop vm first before drop ovs bridge.

Comment (FengYang) [inline qemu/tests/ovs_host_vlan.py]: On 01/13/2015 11:19 AM, Xu Tian wrote:
>
> In qemu/tests/ovs_host_vlan.py
> <https://github.com/autotest/tp-qemu/pull/241#discussion-diff-22841374>:
>
> > +            shell_port = params.get("shell_port", "22")
> > +            data_port = params.get("nc_file_transfer_port", "9000")
> > +            log_file = "file_transfer_from_%s_to_%s.log" % (ips[0], ips[1])
> > +            if os_type == "linux":
> > +                remote.nc_copy_between_remotes(ips[0], ips[1], shell_port,
> > +                                               password, password,
> > +                                               username, username,
> > +                                               file_path, file_path,
> > +                                               d_port=data_port,
> > +                                               timeout=2,
> > +                                               check_sum=True,
> > +                                               s_session=sessions[0],
> > +                                               d_session=sessions[1],
> > +                                               file_transfer_timeout=f_timeout)
> > +    finally:
> > +        ovs_del_cmd = "ovs-vsctl del-br %s" % bridge_name
>
> Do we need to stop vm first before drop ovs bridge.
>
Until now did not find it break anything.  VM should be killed after
this command soon.

File: qemu/tests/ovs_host_vlan.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+            session_ctl = vm.wait_for_serial_login(timeout=login_timeout)
+            txt = "Stop NetworkManager service in guest %s." % vm_name
+            session_ctl.cmd("service NetworkManager stop",
+                            ignore_all_errors=True)
+
+            txt = "Set guest %s IP to %s" % (vm_name, guest_ip)
+            error.context(txt, logging.info)
+            mac = vm.get_mac_address()
+            utils_net.set_guest_ip_addr(session_ctl, mac, guest_ip,
```
Comment (spcui) [inline qemu/tests/ovs_host_vlan.py]: Tested result:
    File "/root/autotest/client/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/ovs_host_vlan.py", line 163, in run
      utils_net.set_guest_ip_addr(session_ctl, mac, guest_ip,
  AttributeError: 'module' object has no attribute 'set_guest_ip_addr'

**Comment by ypu** [comment]
@spcui Hi seems there is a related pullreq in virt-test for this:

https://github.com/autotest/virt-test/pull/1924

Please remember to apply it in your test tree too.

File: qemu/tests/ovs_host_vlan.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            raise error.TestError(err)
+
+        txt = "Run netperf client in VM %s" % vms[1]
+        error.context(txt, logging.info)
+        test_duration = int(params.get("netperf_test_duration", 60))
+        test_protocol = params.get("test_protocol")
+        netperf_cmd = params.get("netperf_cmd")
+        netperf_cmd = os.path.join(netperf_path, netperf_cmd)
+        cmd = netperf_cmd % (test_duration, ips[0])
+        if test_protocol:
```
Comment (xutian) [inline qemu/tests/ovs_host_vlan.py]: @ we have  -t %s in  qemu/tests/cfg/ovs_host_vlan.cfg :line27, so here will get invaild netperf command

**Comment by xutian** [comment]
TestFail: Fail to run netperf test between virt-tests-vm1 and vm2. Command output:
/var/tmp/netperf-2.6.0/src/netperf -l 60 -H 192.168. 10.11
establish control: are you sure there is a netserver listening on 192.168.10.11 at port 12865?
establish_control could not establish the control connection from 0.0.0.0 port 0 address family AF_UNSPEC to 192.168.10.11 port 12865 address family AF_INET

**Pull request closed without merging**